### PR TITLE
Subscription request section fixes

### PR DESCRIPTION
**Pull request by wmaethner**
Fixes to issues: #92, #130, #131, #132, #184, #186, #187, #188, #189, #190

### CODE REVIEW

**Comment by wmaethner** [comment]
@NiklasSvenzen Looks like you fixed that issue, thanks! Anything else that this PR needs?

File: docs/specification/STU1.md [hunk 1]
```diff
@@ -40,24 +40,26 @@ Subscribing consists of two exchanges:
...
+Hubs SHALL allow subscribers to re-request subscriptions that are already activated. Each subsequent and verified request to a hub to subscribe or unsubscribe SHALL override the previous subscription state for a specific topic / callback URL combination.
 
-Within FHIRcast, the client that creates a subscription and the server that hosts the callback url are the same entity. If these roles are split, the Hub assumes that the same authorization and access rights apply to both systems. 
+The callback URL MAY contain arbitrary query string parameters (e.g., ?foo=bar&red=fish). Hubs SHALL preserve the query string during subscription verification by appending new, Hub-defined parameters to the end of the list using the & (ampersand) character to join When sending the event notifications, the hub SHALL make a POST request to the callback URL including any query string parameters in the URL portion of the request, not as POST body parameters.
```
Comment (lbergnehr) [inline docs/specification/STU1.md]: Some minor stuff.

```suggestion
The callback URL MAY contain arbitrary query string parameters (e.g., `?foo=bar&red=fish`). Hubs SHALL preserve the query string during subscription verification by appending new, Hub-defined, parameters to the end of the list using the `&` (ampersand) character to join. When sending the event notifications, the Hub SHALL make a POST request to the callback URL including any query string parameters in the URL portion of the request, not as POST body parameters.
```

File: docs/specification/STU1.md [hunk 2]
```diff
@@ -40,24 +40,26 @@ Subscribing consists of two exchanges:
...
+
+The client that creates the subscription may not be the same system as the server hosting the callback url. (For example, some type of federated authorization model could possibly exist between these two systems). However, in FHIRcast, the hub assumes that the same authorization and access rights apply to both the subscribing client and the callback url.
```
Comment (lbergnehr) [inline docs/specification/STU1.md]:
```suggestion
The client that creates the subscription may not be the same system as the server hosting the callback url. (For example, some type of federated authorization model could possibly exist between these two systems.) However, in FHIRcast, the Hub assumes that the same authorization and access rights apply to both the subscribing client and the callback url.
```

**Comment by wmaethner** [comment]
Thanks @lbergnehr, nice catches, I applied those changes.

**Pull request merged**